src/widgets/sparkline.rs

```diff
@@ -169,38 +183,46 @@ impl Sparkline<'_> {
             .take(max_index)
             .map(|e| {
                 if max == 0 {
-                    0
+                    Some(0)
+                } else if *e == 0 {
+                    None
```

Comment:
Looking at this line in particular, it sort of suggests an idea that this could be more flexible to implement in a way that each data item is Option<u64> instead of treating 0 as a special case. This would allow the Sparkline to indicate both 0 data and missing data points on the same graph. This might be more difficult to work out a good migration strategy for however - likely adding a new method to support this (unless there's a really good way to make the method accept both datasets - I can't see an obvious one though)